src/robot/tidypkg/transformers.py

```diff
@@ -405,6 +412,8 @@ def visit_Statement(self, statement):
         return statement
 
     def _should_be_aligned(self, tokens):
+        if self.space_count == 'preserve':
+            return False
```

Comment:
1. Are you sure disabling alignment is a good idea? It is done only when test case sections has additional headers like

    ```robotframework
    *** Test Cases ***    USER     PWD
    Valid login           robot    secret
    Invalid login         robot    bad
    ```
    and in that case data in columns is aligned consistently. We can argue that explicitly specified headers should always enable this behavior, but then again explicitly used `preserve` should also preserve spaces. I guess not aligning columns in this case is fine.

2. In this class this attribute is used only to control is data aligned or not and thus `space_count` is pretty bad name. It should be changed to something like `no_align` or `preserve_spaces`.

3. The sole purpose of the Aligner class is to align data. If that's not desired, it would actually be better to not use the Aligner at all.The current

    ```python
    Aligner(...).visit(model)
    ```
    in `Tidy._tidy` should be changed to
    ```python
    if self.space_count != 'preserve':
        Aligner(...).visit(model)
    ```

Reply:
First of all, sorry for the long comment 😉

For this sample code:
```
*** Test Cases ***      USERNAME        PASSWORD
Valid Login             robot           secret
Invalid Login           robot           wrong
```
I get this, when `Aligner` is enabled and my changes from this PR are included:
```
*** Test Cases ***      USERNAME        PASSWORD
Valid Login           robot              secret

Invalid Login         robot              wrong

```
I get this, when `Aligner` is disabled for `preserve` value of `spacecount` (according to your latest suggestion):
```
*** Test Cases ***      USERNAME        PASSWORD
Valid Login    robot           secret

Invalid Login    robot           wrong

```
So the templated test cases are ignored entirely when I want to preserve spaces which is not the intended solution.

And I get this, when not using `spacecount` option at all (so basically this is how it works on master right now):
```
*** Test Cases ***    USERNAME    PASSWORD
Valid Login           robot       secret

Invalid Login         robot       wrong

```
**The problem is** that every time there is a new line after each templated test case. I tried to debug this but I couldn't find the cause of this issue. What I found is that inside each `TestCase` object body there are 2 other objects: `TemplateArguments` and `EmptyLine` but there is also an `EOL` token at the end of the `TemplateArguments` so the **new line is duplicated**. Visualization:
```
Valid Login           robot       secret  # TemplateArguments(SEPARATOR, 'robot', 'SEPARATOR', 'secret', '\r\n') object
# EmptyLine object
Invalid Login         robot       wrong # TemplateArguments(SEPARATOR, 'robot', 'SEPARATOR', 'wrong', '\r\n') object
# EmptyLine object
```

Whatever I do, the problem is there, regardless of the suggestions to this PR. The best results I get with using tidy when I keep the current changes from this PR.

**What do I propose?**
We have different options:
- Merge the changes from this PR like they are
- Debug the issue, fix it and then merge (I cannot promise I will do it)
- Discard the changes and forget about them
- Rewrite the tidy as a new external tool
I suggest merging this PR and giving a chance for the last one: rewriting the tidy.

**Why to rewrite the tidy?**
- Keeping such tools (like tidy) as a part of the RF package may lead to maintenance nightmare - by removing them we don't need to worry about implementing the latest changes to RF in all internal tools
- It can be better - it has some flaws and issues for some specific situations
- Maintaining of RF would be cheaper because the tool will be stored as an external package
- Many more opportunities e.g. we can introduce simple configuration files which will give the developers the power of configuring the formatting for each project

**Who will do it?**
I can volunteer on it. And of course I am open to any help 😀 I already have some experience with new robot API (gained when creating [Robocop](https://github.com/MarketSquare/robotframework-robocop)) and I know how much such tools are helpful in my current projects where several developers create tests and they are struggling to follow the guidelines - refactoring is sometimes a very time-consuming task.